examples/README.md

```diff
@@ -17,3 +17,12 @@ cargo run --package example-{name of example}
 ## Full Example
 
 Look at the [full-api](./full-api) example to see autometrics in use in an example API server built with `axum`, `thiserror`, and `tokio`.
+
+## Prometheus
+
+If Prometheus is installed locally, the examples may start it and configure it to scrape the example servers. You can also start Prometheus before running the examples with Docker like this:
+
+```
+# from this directory
+docker run  -it -v $(pwd)/util/prometheus.yml:/prometheus/prometheus.yml --network host prom/prometheus --enable-feature=exemplar-storage
```

Comment:
The `--network host` didn't work for me on Mac but replacing that with the following did work

```suggestion
docker run  -it -v $(pwd)/util/prometheus.yml:/prometheus/prometheus.yml -p 9090:9090 prom/prometheus --enable-feature=exemplar-storage
```

Reply:
Yes this will work for prometheus port, but --network host is there for prometheus to be able to connect to localhost, where example is publishing its metrics. 

I guess my approach is focusing on linux, on other OSes it might have use if prometheus is already running as native process.

Reply:
Gotcha, that makes sense.

I think if we have instructions for running it in Docker, it would be best if they worked across OSes, though I'm not entirely sure what we'd need to do to make that work.

Reply:
I'm do not have experiences with Docker Desktop on Win nor MacOS.   In both cases it I think it runs in VM, so there is no chance to connect to  localhost where example is exposing metrics.

However there is `host.docker.internal` internal DNS record which works for MacOS and Win - but [not exactly the same for Linux](https://stackoverflow.com/questions/48546124/what-is-linux-equivalent-of-host-docker-internal).

I think there are many options how one can potentially run examples:
- with locally installed Prometheus and examples will start/stop it - current scenario
- with locally installed Prometheus, user will start it in advance -  here the PR will help on all OS,
- on Linux with Prometheus running in container  - again this works with this PR
- Win, MacOS (Docker Desktop)- it'll require more modification - first not trying to start Prometheus (env. variable?) , second change configuration localhost -> host.docker.internal 
- Prometheus running on different host?

Question is if this should be captured in examples logic, or just add bit more description to readme.

Definitely containerization of dependencies/services  it trend for testing - [TestContainers](https://testcontainers.com/), so this would be also way for examples, unfortunately Prometheus has an "unlucky" feature, that instead of listening on port like majority servers do, it also needs to connect to the example.  So I did not find any useful inspiration in TestContainers. 

So options I see them now:
- extend a bit readme
- or close PR as this is not general enough - Linux only